rita_common/src/payment_validator/mod.rs

```diff
@@ -788,15 +789,17 @@ fn handle_tx_messaging(transaction: TransactionDetails, ts: ToValidate, current_
                 );
             }
 
-            // update debt keeper with the details of this payment
-            let _ = payment_received(
-                pmt.from,
-                pmt.amount,
-                denom.expect("How did this happen when we already verified existence"),
-            );
+            if !payment_accounted_for(pmt.clone()) {
+                // update debt keeper with the details of this payment
+                let _ = payment_received(
```

Comment:
call payment accounted for inside update payments, but send it off to the payment validator, part of hte goal of re-sending payment data was to ensure routers didn't enforce on each other, right now the only bug we're trying to fix is duplicate data in the earnings display, which won't be affected by us applying a payment twice due to the way the internals of debt keeper work.